feature/experimental/bgp/ate_tests/bgp_tcp_mss_path_mtu/bgp_tcp_mss_path_mtu_test.go

```diff
@@ -0,0 +1,406 @@
+		isisIntf.Enabled = ygot.Bool(true)
+		isisIntf.CircuitType = oc.Isis_CircuitType_POINT_TO_POINT
+		isisIntfLevel := isisIntf.GetOrCreateLevel(2)
+		isisIntfLevel.Enabled = ygot.Bool(true)
+		isisIntfLevelAfi := isisIntfLevel.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST)
+		isisIntfLevelAfi.Metric = ygot.Uint32(200)
+		isisIntfLevelAfi.Enabled = ygot.Bool(true)
+	}
+	gnmi.Replace(t, dut, dutConfIsisPath.Config(), prot)
+}
+
+// configureATE configures the interfaces and BGP protocols on an ATE, including
+// advertising some(faked) networks over BGP.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	topo := ate.Topology().New()
+
+	port1 := ate.Port(t, "port1")
+	iDut1 := topo.AddInterface(atePort1.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(atePort1.IPv4CIDR()).WithDefaultGateway(dut1Port1.IPv4)
+	iDut1.IPv6().WithAddress(atePort1.IPv6CIDR()).WithDefaultGateway(dut1Port1.IPv6)
+
+	isisDut1 := iDut1.ISIS()
+	isisDut1.WithLevelL2().WithNetworkTypePointToPoint().WithTERouterID(atePort1.IPv4)
+
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dut1Port1.IPv4).WithLocalASN(ateAS1).
+		WithTypeExternal()
+	bgpDut1.AddPeer().WithPeerAddress(dut1Port1.IPv6).WithLocalASN(ateAS1).
+		WithTypeExternal()
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+
+	topo.Push(t)
+	topo.StartProtocols(t)
+	return topo
+}
+
+func configATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+
+	topo := ate.Topology().New()
+
+	port1 := ate.Port(t, "port1")
+	iDut1 := topo.AddInterface(atePort1.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(atePort1.IPv4CIDR()).WithDefaultGateway(dut1Port1.IPv4)
+	iDut1.IPv6().WithAddress(atePort1.IPv6CIDR()).WithDefaultGateway(dut1Port1.IPv6)
+
+	isisDut1 := iDut1.ISIS()
+	isisDut1.WithLevelL2().WithNetworkTypePointToPoint().WithTERouterID(atePort1.IPv4)
+
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dut2Port1.IPv4).WithLocalASN(ateAS1).
+		WithTypeInternal().WithMD5Key(authPWd)
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+
+	topo.Push(t)
+	topo.StartProtocols(t)
+	return topo
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string) {
+	ni := oc.NetworkInstance{Name: ygot.String(deviations.DefaultNetworkInstance(dut))}
+	static := ni.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+	sr := static.GetOrCreateStatic(prefix)
+	nh := sr.GetOrCreateNextHop("0")
+	nh.NextHop = oc.UnionString(nexthop)
+	gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut)).Config(), static)
+}
+
+// TestTcpMssPathMtu is to Validate TCP MSS for BGP v4/v6 sessions.
+func TestTcpMssPathMtu(t *testing.T) {
```

Comment:
per go style should be named "TestTCPMSSPathMTU"

Reply:
updated.